src/java.desktop/share/classes/javax/swing/JTabbedPane.java

```diff
@@ -2339,13 +2339,16 @@ public boolean contains(Point p) {
         public Point getLocationOnScreen() {
              Point parentLocation = parent.getLocationOnScreen();
```

Comment:
possibly the "parent.getLocationOnScreen()" should be wrapped by the "try/catch IllegalComponentStateException"? If some parent is not visible then that exception will be thrown, but per the spec, the null should be returned.

I think this could be covered by the test since the change is in the public shared code.

Reply:
Sure. I have added the handler ICSE and a regression test.